src/game/rooms.js

```diff
@@ -1130,8 +1145,30 @@ exports.make = function(_runtimeData, _intents, _register, _globals) {
         globals.console.clearVisual(this.roomName);
         return this;
     });
-
+    
     Object.defineProperty(globals, 'RoomVisual', {enumerable: true, value: RoomVisual});
+
+    
+    Room.Terrain = register.wrapFn(function(roomName){ "use strict";
+        roomName = "" + roomName;
+        
+        const array = (runtimeData.staticTerrainData || {})[roomName];
+        if(!array)
+            throw new Error(`Could not access room ${roomName}`);
+        
+        this.get = register.wrapFn(function(x,y){
+            const value = array[y * 50 + x];
+            return (value & C.TERRAIN_MASK_WALL) || (value & C.TERRAIN_MASK_SWAMP) || 0;
+        });
+        
+        this.getRawBuffer = register.wrapFn(function(destinationArray){
+            if(!!destinationArray) {
+                TerrainConstructorSet.call(destinationArray, array);
+                return destinationArray;
+            }
+            return new TerrainConstructor(array);
```

Comment:
I think, `TerrainConstructor` and `TerrainConstructorSet` need to be wrapped with `wrapFn` as well, but not sure. Smth like:

```js
return new (register.wrapFn(TerrainConstructor)(array));
```